pygmt/src/config.py

```diff
@@ -20,6 +22,154 @@ class config:  # pylint: disable=invalid-name
+        "FORMAT_TIME_SECONDARY_MAP",
+        "FORMAT_TIME_STAMP",
+        "GMT_DATA_SERVER",
+        "GMT_DATA_SERVER_LIMIT",
+        "GMT_DATA_UPDATE_INTERVAL",
+        "GMT_COMPATIBILITY",
+        "GMT_CUSTOM_LIBS",
+        "GMT_EXPORT_TYPE",
+        "GMT_EXTRAPOLATE_VAL",
+        "GMT_FFT",
+        "GMT_GRAPHICS_DPU",
+        "GMT_GRAPHICS_FORMAT",
+        "GMT_HISTORY",
+        "GMT_INTERPOLANT",
+        "GMT_LANGUAGE",
+        "GMT_MAX_CORES",
+        "GMT_THEME",
+        "GMT_TRIANGULATE",
+        "GMT_VERBOSE",
+        "IO_COL_SEPARATOR",
+        "IO_FIRST_HEADER",
+        "IO_GRIDFILE_FORMAT",
+        "IO_GRIDFILE_SHORTHAND",
+        "IO_HEADER",
+        "IO_HEADER_MARKER",
+        "IO_N_HEADER_RECS",
+        "IO_NAN_RECORDS",
+        "IO_NC4_CHUNK_SIZE",
+        "IO_NC4_DEFLATION_LEVEL",
+        "IO_LONLAT_TOGGLE",
+        "IO_SEGMENT_BINARY",
+        "IO_SEGMENT_MARKER",
+        "MAP_ANNOT_MIN_ANGLE",
+        "MAP_ANNOT_MIN_SPACING",
+        "MAP_ANNOT_OBLIQUE",
+        "MAP_ANNOT_OFFSET_PRIMARY",
+        "MAP_ANNOT_OFFSET_SECONDARY",
+        "MAP_ANNOT_ORTHO",
+        "MAP_DEFAULT_PEN",
+        "MAP_DEGREE_SYMBOL",
+        "MAP_EMBELLISHMENT_MODE",
+        "MAP_FRAME_AXES",
+        "MAP_FRAME_PEN",
+        "MAP_FRAME_PERCENT",
+        "MAP_FRAME_TYPE",
+        "MAP_FRAME_WIDTH",
+        "MAP_GRID_CROSS_SIZE_PRIMARY",
+        "MAP_GRID_CROSS_SIZE_SECONDARY",
+        "MAP_GRID_PEN_PRIMARY",
+        "MAP_GRID_PEN_SECONDARY",
+        "MAP_HEADING_OFFSET",
+        "MAP_LABEL_MODE",
+        "MAP_LABEL_OFFSET",
+        "MAP_LINE_STEP",
+        "MAP_LOGO",
+        "MAP_LOGO_POS",
+        "MAP_ORIGIN_X",
+        "MAP_ORIGIN_Y",
+        "MAP_POLAR_CAP",
+        "MAP_SCALE_HEIGHT",
+        "MAP_TICK_LENGTH_PRIMARY",
+        "MAP_TICK_LENGTH_SECONDARY",
+        "MAP_TICK_PEN_PRIMARY",
+        "MAP_TICK_PEN_SECONDARY",
+        "MAP_TITLE_OFFSET",
+        "MAP_VECTOR_SHAPE",
+        "PROJ_AUX_LATITUDE",
+        "PROJ_DATUM",
+        "PROJ_ELLIPSOID",
+        "PROJ_GEODESIC",
+        "PROJ_LENGTH_UNIT",
+        "PROJ_MEAN_RADIUS",
+        "PROJ_SCALE_FACTOR",
+        "PS_CHAR_ENCODING",
+        "PS_COLOR_MODEL",
+        "PS_COMMENTS",
+        "PS_CONVERT",
+        "PS_IMAGE_COMPRESS",
+        "PS_LINE_CAP",
+        "PS_LINE_JOIN",
+        "PS_MITER_LIMIT",
+        "PS_MEDIA",
+        "PS_PAGE_COLOR",
+        "PS_PAGE_ORIENTATION",
+        "PS_SCALE_X",
+        "PS_SCALE_Y",
+        "PS_TRANSPARENCY",
+        "TIME_EPOCH",
+        "TIME_IS_INTERVAL",
+        "TIME_INTERVAL_FRACTION",
+        "TIME_LEAP_SECONDS",
+        "TIME_REPORT",
+        "TIME_UNIT",
+        "TIME_WEEK_START",
+        "TIME_Y2K_OFFSET_YEAR",
+    ] + [  # special keywords
+        "FONT",
+        "FONT_ANNOT",
+        "FORMAT_TIME_MAP",
+        "MAP_ANNOT_OFFSET",
+        "MAP_GRID_CROSS_SIZE",
+        "MAP_GRID_PEN",
+        "MAP_TICK_LENGTH",
+        "MAP_TICK_PEN",
+    ]
```

Comment:
I think it makes more sense to also put the `self.special_params` attribute out of the `__init__` function.

Reply:
Yeah, I was thinking about that too. Maybe create a `_special_params` variable to hold that dictionary.

Reply:
Done in 318501bf8d870278a5199649b93bd31a18d84f71, and the variable name is `_special_keywords` instead of `_special_params`.

Reply:
Nice work, just double checked one by one and it looks good!